cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -477,6 +491,87 @@ class ArrayCountSorter {
   }
 };
 
+using ::arrow::internal::Bitmap;
+
+template <>
+class ArrayCountSorter<BooleanType> {
+ public:
+  ArrayCountSorter() = default;
+
+  // Returns where null starts.
+  uint64_t* Sort(uint64_t* indices_begin, uint64_t* indices_end,
+                 const BooleanArray& values, int64_t offset,
+                 const ArraySortOptions& options) {
+    // 32bit counter performs much better than 64bit one
+    if (values.length() < (1LL << 32)) {
+      return SortInternal<uint32_t>(indices_begin, indices_end, values, offset, options);
+    } else {
+      return SortInternal<uint64_t>(indices_begin, indices_end, values, offset, options);
+    }
+  }
+
+ private:
+  template <typename CounterType>
+  void CountBits(const BooleanArray& values, int64_t offset, CounterType* set_count,
+                 CounterType* null_count) {
```

Comment:
Really, you needn't write this yourself. Just call `null_count()` and `true_count()`.

Reply:
Oh okay. I did this way because we can count both nulls and trues in a single pass. But sure, I'll use that.